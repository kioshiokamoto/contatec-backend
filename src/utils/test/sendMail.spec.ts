import sendEmail from '../sendMail';
let email = sendEmail;
email = jest.fn(() => 'Autorizado');

const functionEmail = {
  sendEmail,
};

describe('Utilidad de envio de correos', () => {
  it('Debe ser equivalente a codigo generado', () => {
    expect(sendEmail).toMatchSnapshot();
  });
  it('Debe poder ser llamado', () => {
    const result = email('test@mail.com', 'http://google.com', 'test');
    expect(result).toBe('Autorizado');
    expect(email).toHaveBeenCalled();
    expect(email).toHaveBeenCalledWith(
      'test@mail.com',
      'http://google.com',
      'test',
    );
  });
  it('Debe poder ser llamado', () => {
    const spy = jest.spyOn(functionEmail, 'sendEmail');

    const sendingEmail = functionEmail.sendEmail(
      'test@mail.com',
      'http://google.com',
      'test',
    );

    expect(spy).toHaveBeenCalled();
    //No retorna nada
    expect(sendingEmail).toBe(undefined);
    spy.mockReset();
    spy.mockRestore();
  });
});
