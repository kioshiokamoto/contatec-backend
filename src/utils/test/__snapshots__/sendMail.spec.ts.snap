// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Utilidad de envio de correos Debe ser equivalente a codigo generado 1`] = `[Function]`;
